### PR TITLE
Working cookbook - creates masters and nodes, and tests successfully creating a pod and service

### PR DESCRIPTION
**Pull request by adamleff**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,27 +20,87 @@ TODO: List your supported platforms.
...
+  master_ip '192.168.122.50'
+  role :node
+  action :create
+end
+```
+
+The following attributes are accepted in the `atomic_host` resource:
+
+ * `ip_address`: **required** - IP address to assign to the Atomic host.  You should ensure this is an IP in an available subnet in your virtualization setup (i.e. libvirt's default of 192.168.122.0/24)
+ * `role`: **required** - should be either `:master` or `:node`
```
Comment (ryancragun) [inline README.md]: What is the default?

File: README.md [hunk 2]
```diff
@@ -16,27 +20,87 @@ TODO: List your supported platforms.
...
+ * `master_ip`: **required if role == :node** - the IP address of the master.  The node will not be able to function without knowing which master to join.
```
Comment (ryancragun) [inline README.md]: Is it possible to use FQDNs and IPs?

File: README.md (hunk 1 above)
Comment (adamleff) [inline README.md]: I opted to not specify a default, making the user choose it as a required param.  I welcome a differing opinion.

File: libraries/helpers.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+    source_file_md5 = Digest::MD5.file(atomic_file_local_path(ip_address, file))
+
+    dest_file_cmd = run_cmd_on_atomic_host(ip_address, "md5sum #{file}")
+    dest_file_md5 = dest_file_cmd.stdout.split(' ')[0]
+
+    source_file_md5.to_s != dest_file_md5
+  end
+
+  def run_cmd_on_atomic_host(ip_address, command, timeout_duration=nil)
```
Comment (ryancragun) [inline libraries/helpers.rb]: Why do you just set 30 as the default?

File: README.md (hunk 2 above)
Comment (adamleff) [inline README.md]: There's nothing in my code that prevents it, it's just on the user to ensure DNS resolves properly, which may open a can of worms for us to manage the resolv.conf on these hosts.

File: libraries/helpers.rb (hunk 3 above)
Comment (adamleff) [inline libraries/helpers.rb]: General arbitrariness based on testing of some of these commands.  No reason it couldn't be smaller, larger, or required on every method call.  I just picked something, and I welcome suggestions :)

File: libraries/helpers.rb [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    cmd = run_cmd_on_atomic_host(ip_address, command, timeout_duration)
+    cmd.error!
+  end
+
+  def cmd_on_atomic_host_success?(ip_address, command, timeout_duration=nil)
+    cmd = run_cmd_on_atomic_host(ip_address, command, timeout_duration)
+    ! cmd.error?
+  end
+
+  def random_password
```
Comment (ryancragun) [inline libraries/helpers.rb]: Super nit picky, but SecureRandom is probably a better that Random

Comment (adamleff) [inline libraries/helpers.rb]: I :heart: nit-picky stuff.  I'll change it out.

File: libraries/provider_atomic_host.rb [hunk 5]
```diff
@@ -1,20 +1,55 @@
...
-        base_image = ::File.join(node['atomic']['work_dir'], node['atomic']['image_version']).sub('qcow2.xz', 'qcow2')
-        init_iso = "#{node['atomic']['work_dir']}/#{new_resource.instance_id}/init.iso"
-        resource_dir = "#{node['atomic']['work_dir']}/#{new_resource.instance_id}"
-        meta_data_file = "#{resource_dir}/meta-data"
-        user_data_file = "#{resource_dir}/user-data"
+
+        ip_address  = new_resource.ip_address
+        mac_address = kvm_mac_by_ip(ip_address)
+        role        = new_resource.role.to_sym
+        password    = new_resource.password.nil? ? (0...8).map { (65 + rand(26)).chr }.join : new_resource.password
```
Comment (ryancragun) [inline libraries/provider_atomic_host.rb]: Should probably use the password helper here

Comment (adamleff) [inline libraries/provider_atomic_host.rb]: Yup - bug.  thanks for catching.

File: libraries/provider_atomic_host.rb [hunk 6]
```diff
@@ -23,34 +58,251 @@ def whyrun_supported?
...
+          ruby_block "#{ip_address} create local-registry container" do
+            block do
+              run_cmd_on_atomic_host!(ip_address, docker_cmd.join(' '), timeout_duration=300)
+            end
+            action :run
+            not_if { cmd_on_atomic_host_success?(ip_address, "docker inspect local-registry") }
+          end
+
+          atomic_file "#{ip_address} systemd local-registry" do
+            ip_address ip_address
```
Comment (ryancragun) [inline libraries/provider_atomic_host.rb]: I'm not a huge fan of variable names shadowing method names.

Comment (adamleff) [inline libraries/provider_atomic_host.rb]: That's a fair point - it does look funny.  I'll address. :+1:

File: libraries/helpers.rb (hunk 3 above)
Comment (ryancragun) [inline libraries/helpers.rb]: My point is if you set timeout_duration = 30 in the method definition you don't need to set a default on the next line.

**Comment by ryancragun** [comment]
:+1: After the ip_address change.  Super badass work.

**Pull request merged**